pkg/jobworker/concurrent_test.go

```diff
@@ -0,0 +1,62 @@
+package jobworker
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"testing"
+)
+
+const numClients = 10
+
+func TestConcurrentReaders(t *testing.T) {
+	WORKER_UID = -1
+	WORKER_GUID = -1
+	// Define number of log iterations and content
+	n := 5
+	echo := "hello"
+	// Define job's command and options for test
+	cmd := "bash"
+	args := []string{"-c", fmt.Sprintf("for run in {1..%d}; do echo %s; sleep 0.1; done", n, echo)}
+	opts := NewOpts(100, 50, 100*CgroupMB)
+	// Run the job
+	job, err := StartWithController(&mockController{}, opts, cmd, args...)
+	if err != nil {
+		fmt.Print("failed to start command")
+		fmt.Print(err)
+		return
+	}
+
+	// Spawn multiple clients to read log output
+	for _ = range numClients {
+		// Get io.ReadCloser tailing job logs
+		reader, err := job.Output(false)
+		if err != nil {
+			t.Errorf("could not get reader for job's output: %v", err)
+			return
+		}
+		defer reader.Close()
+
+		// Read logs and assert contents and length
```

Comment:
As far as I can see you're not really asserting either of those things.

This test just stops counting after it gets to _n_ lines of log output. It counts getting any arbitrary number lines as a success, as long as all of those log lines contain "hello". You could have no output at all and this test would pass.

You probably want to add a line count to the output (see above) and rewrite the assertion to look something like:

```golang
go func(r io.ReadCloser) {
	scanner := bufio.NewScanner(r)
	logs := []string{}
	for scanner.Scan() {
		logs = append(logs, scanner.Text())
	}

	if len(logs) != n {
		t.Failf("Expected %d lines, got %d", n, len(logs)
	}

	for i, log := range logs {
		expected := fmt.Sprintf("%d: %s", i+1, echo)
		if log != expected {
			t.Failf("Line %d, expected %s, got %s", i, expected, log)
		}
	}
}(reader)
```